cmd/agent/subcommands/run/command.go

```diff
@@ -465,6 +465,9 @@ func startAgent(
 	// Detect Cloud Provider
 	go cloudproviders.DetectCloudProvider(context.Background())
 
+	// Collect account ID for inventories
+	go cloudproviders.CollectAccountID(context.Background())
+
```

Comment:
This will be done automatically by the inventory payload, we should not rely on the init sequence from the main.